packages/flutter/lib/src/widgets/conduit.dart

```diff
@@ -0,0 +1,375 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' show max;
+
+import 'package:flutter/foundation.dart' show immutable;
+import 'package:flutter/rendering.dart';
+
+import 'framework.dart';
+
```

Comment:
Let's add a comment at the top of this file that explains the general layout, since it's a bit subtle. As I understand it, it's:

```
//      widget: Conduit; has child, remoteChild, remoteKey (an AnchorKey)
//      element: ConduitElement; inflates child and remoteChild, connects its RO to the remoteKey's RO
//      render object: _RenderLeaderProxyBox; connects to a _RenderFollowerProxyBox
// 
//      render object: _RenderFollowerProxyBox; connects to a _RenderLeaderProxyBox
//      element: _RenderConduitAnchorElement
//      widget: RenderConduitAnchorWidget, whose key is an AnchorKey
```

Reply:
This also suggests we should probably clean up the names. In particular, "RenderConduitAnchorWidget" is an impossible name since it suggests the widget is a render object.

Reply:
(also, `_RenderFollowerProxyBox` and `_RenderConduitAnchorElement` don't sound related at all)

Reply:
e.g.

widget: Conduit
element: ConduitElement
renderObject: RenderConduit

widget: ConduitAnchor
element: ConduitAnchorElement
renderObject: RenderConduitAnchor